### DISCUSSION TITLE
How to create a serverless function cron job with Node.js?

### DISCUSSION THREAD

**Discussion by mattbal** [vercel/vercel#11798] [Help]
### Question

I created a Node.js project that I want to run as a serverless function cron job, but when I deployed it to production on Vercel, the cron job didn't run.

Can anyone take a look at my code and tell me what I'm doing wrong? Am I defining the serverless function correctly?

https://github.com/mattbal/DailyZenQuotes/tree/main/sendmail

**Comment by mattbal** [comment]
bump

**Comment by amyegan** [comment]
Hey @mattbal. If you already have a serverless function created and you can confirm that it's working, then you just need to configure the cron job in a `vercel.json` config file. You'll use the [crons](https://vercel.com/docs/projects/project-configuration#crons) option to set the path of the deployed serverless function and a cron expression for the schedule. Keep in mind that Hobby plans are limited to 2 cron jobs each triggered once per day.

https://vercel.com/guides/how-to-setup-cron-jobs-on-vercel

If you still need to get the serverless function working, I recommend putting the function code in a `/api` directory at the root of the project. Here are some docs with more info.
- [Node.js Runtime for Serverless Functions](https://vercel.com/docs/functions/runtimes/node-js)
- [Vercel Functions Quickstart](https://vercel.com/docs/functions/quickstart#vercel-functions-quickstart)
- [Using Express.js with Vercel](https://vercel.com/docs/functions/runtimes/node-js) (has helpful setup tips even if you're not using Express)
